smart-py/marketplace.py

```diff
@@ -1,65 +1,954 @@
+            amount=sp.TNat,
+            token_id=sp.TNat,
+            token_info=sp.TMap(sp.TString, sp.TBytes)
+        ),
+        objkt.address,
+        entry_point = "mint"
+    ).open_some()
+    #
+    # objktTransfer = sp.transfer(
+    #     sp.record(
+    #         address=creator.address,
+    #         amount=1,
+    #         token_id=153,
+    #         token_info={ '' : sp.bytes('0x697066733a2f2f516d61794e7a7258547a354237577747574868314459524c7869646646504676775a377a364b7443377268456468') }
+    #     ),
+    #     sp.mutez(0),
+    #     objktContract
+    # )
+
+    # add an 1/1 objkt to the contract
+    # the address and the sender are both the creator
+    # scenario += swap.mint_OBJKT(
+    #     address = creator.address,
+    #     amount = 1,
+    #     royalties = 200,
+    #     metadata = sp.bytes('0x697066733a2f2f516d61794e7a7258547a354237577747574868314459524c7869646646504676775a377a364b7443377268456468')
+    # ).run(
+    #     sender = creator.address,
+    #     valid = True
+    # )
+    #
+    # # the mint was successful but still no swap
+    # scenario.verify(swap.data.objkt_id == 153)
+    # scenario.verify(swap.data.swap_id == 0)
+    # scenario.verify(swap.data.swaps.contains(0) == False)
+
+    # try swap more objkts than exist must fail
+    # TODO this should fail and currently does not
+    # scenario += swap.swap(
+    #     objkt_id = 153,
+    #     objkt_amount = 2,
+    #     xtz_per_objkt = sp.utils.nat_to_mutez(1)
+    # ).run(
+    #     sender = seller.address,
+    #     valid = False
+    # )
+
+    # swap id should not have incremented
+    # scenario.verify(swap.data.swaps.contains(0) == False)
+
+    # swap with new objkt id must pass
+    scenario += swap.swap(
+        creator=creator.address,
+        royalties=250,
+        objkt_id = 153,
+        objkt_amount = 1,
+        xtz_per_objkt = sp.utils.nat_to_mutez(1)
+    ).run(
+        sender = seller.address,
+        valid = True
+    )
+
+    # one swap was added
+    scenario.verify(swap.data.swaps.contains(500000) == True)
+    scenario.verify(swap.data.swaps.contains(500001) == False)
+    scenario.verify(swap.data.counter == 500001)
+    scenario.verify(swap.data.swaps.get(500000).objkt_id == 153)
+
+    # swap should now fail because there is only 1 objkt available
+    # TODO this currently passes and a swap gets added breaking all
+    # of the tests that have been commented out below
+    #
+    # tests below may be making incorrect assumptions, do not assume they
+    # are correct because i haven't been able to continue due to this
+    # first bug
+    #
+    # scenario += swap.swap(
+        # objkt_id = 153,
+        # objkt_amount = 1,
+        # xtz_per_objkt = sp.utils.nat_to_mutez(2)
+    # ).run(sender = seller.address, valid = False)
+    #
+    # # swap should not have been added because the swap should not
+    # exist anymore but it gets added
+    #
+    # scenario.verify(swap.data.swap_id == 1)
```

Comment:
Where does the swap_id value get set?

Reply:
that gets set in the objky_swap.py contract

the goal here is to try load details from the other contract here to validate but i'm struggling with that a bit

Reply:
Oh. I thought this was going to be a replacement of objkt_swap.py. Why can't the testing be added to objkt_swap.py?

Reply:
if that's the case then objkt mints are missing from the marketplace.py contract

i was under the impression they were different, so my question around them each having their own swap entrypoint

@crzypatchwork can you clarify here please?